File: LMS_CMS_Backend/LMS_CMS_DAL/Models/Domains/HR/VacationEmployee.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LMS_CMS_DAL.Models.Domains.HR
{
    public class VacationEmployee : AuditableEntity
    {
        [Key]
        public long ID { get; set; }
        public DateOnly Date { get; set; }
        public string? Notes { get; set; }
        public bool HalfDay { get; set; }
        public DateOnly DateFrom { get; set; }
        public DateOnly? DateTo { get; set; }

        [ForeignKey("Employee")]
        public long EmployeeID { get; set; }
        public Employee Employee { get; set; }

        [ForeignKey("VacationTypes")]
        public long VacationTypesID { get; set; }
        public VacationTypes VacationTypes { get; set; }

    }
}

File: LMS_CMS_Frontend/src/app/Models/HR/vacation-employee.ts
export class VacationEmployee {
  constructor(
    public id: number = 0,
    public date: string = '',
    public notes: string = '',
    public dateFrom: string = '',
    public dateTo: string | null= '',
    public halfDay: boolean = false,
    public used: number = 0,
    public balance: number = 0,
    public remains: number = 0,
    public employeeID: number = 0,
    public employeeEnName: string = '',
    public employeeArName: string = '',
    public vacationTypesID: number = 0,
    public vacationTypesName: string = ''
  ) {}
}

